File: frontend/app/pages/CustomerListPage/domain/Customer.repository.ts
import type {CreateCustomer, Customer} from '~/shared/domain/Customer.model';

export interface CustomerRepository {
    loadCustomers(): Promise<Customer[]>;
    findCustomerById(customerId: number): Promise<Customer | undefined>;
    updateCustomer(customer: Customer): Promise<Customer | undefined>;
    deleteByCustomerId(customerId: number): Promise<void>;
    createCustomer(customer: CreateCustomer): Promise<Customer>;
}

export namespace CustomerRepository {
    export const type = Symbol.for('CustomerRepository');
}

File: backend/ui-backend/src/main/java/com/example/demo/domain/CustomerRepository.java
package com.example.demo.domain;

import com.example.demo.domain.model.Customer;
import org.springframework.web.bind.annotation.CrossOrigin;

import java.util.List;

@CrossOrigin
public interface CustomerRepository {
    List<Customer> loadCustomers();

    Customer createCustomer(Customer customer);

    void deleteCustomerById(Integer id);

    Customer getCustomerById(Integer id);
}
